api/test/OpenTelemetry/Logging/CoreSpec.hs

```diff
@@ -0,0 +1,107 @@
+  describe "setGlobalLoggerProvider" $ do
+    it "works" $ do
+      lp <-
+        createLoggerProvider $
+          LoggerProviderOptions
+            { loggerProviderOptionsResource =
+                materializeResources $
+                  toResource
+                    OperatingSystem
+                      { osType = "exampleOs"
+                      , osDescription = Nothing
+                      , osName = Nothing
+                      , osVersion = Nothing
+                      }
+            , loggerProviderOptionsAttributeLimits =
+                LA.AttributeLimits
+                  { attributeCountLimit = Just 50
+                  , attributeLengthLimit = Just 50
+                  }
+            }
+      setGlobalLoggerProvider lp
+      glp <- getGlobalLoggerProvider
+      glp `shouldBe` lp
+  describe "addAttribute" $ do
+    it "works" $ do
+      lp <- getGlobalLoggerProvider
+      let l = makeLogger lp InstrumentationLibrary {libraryName = "exampleLibrary", libraryVersion = "", librarySchemaUrl = "", libraryAttributes = A.emptyAttributes}
+      lr <-
+        emitLogRecord
+          l
+          LogRecordArguments
+            { timestamp = Nothing
+            , observedTimestamp = Nothing
+            , context = Nothing
+            , severityText = Nothing
+            , severityNumber = Nothing
+            , body = Nothing
+            , attributes =
+                H.fromList
+                  [ ("something", "a thing")
+                  ]
+            }
+      addAttribute lr "anotherThing" ("another thing" :: LA.AnyValue)
+
+      (_, attrs) <- LA.getAttributes <$> logRecordGetAttributes lr
+
+      attrs
+        `shouldBe` H.fromList
+          [ ("anotherThing", "another thing")
+          , ("something", "a thing")
+          ]
+  describe "addAttributes" $ do
+    it "works" $ do
+      lp <- getGlobalLoggerProvider
+      let l = makeLogger lp InstrumentationLibrary {libraryName = "exampleLibrary", libraryVersion = "", librarySchemaUrl = "", libraryAttributes = A.emptyAttributes}
+      lr <-
+        emitLogRecord
+          l
+          LogRecordArguments
+            { timestamp = Nothing
+            , observedTimestamp = Nothing
+            , context = Nothing
+            , severityText = Nothing
+            , severityNumber = Nothing
+            , body = Nothing
+            , attributes =
+                H.fromList
+                  [ ("something", "a thing")
+                  ]
+            }
+      addAttributes lr $
+        H.fromList
+          [ ("anotherThing", "another thing" :: LA.AnyValue)
+          , ("twoThing", "the second another thing")
+          ]
+
+      (_, attrs) <- LA.getAttributes <$> logRecordGetAttributes lr
+
+      attrs
+        `shouldBe` H.fromList
+          [ ("anotherThing", "another thing")
+          , ("something", "a thing")
+          , ("twoThing", "the second another thing")
+          ]
```

Comment:
The `it` clauses in this document are rather long, so I wonder if the tests could be separated into smaller parts. As it is, if a test fails you would have to muck around in these lines of code to find which line failed, but at least you will know something is wrong.

Reply:
They are long, but it does seem like one behavior is being described and expected/asserted, which is good.

I could see there being a shorter way to construct `LogRecordArguments`... essentially a `emptyLogRecordArguments` that you use the record modification syntax on like:

```
emptyLogRecordArguments { attributes = H.fromList [ ... ] }
```

Reply:
Done. The way I did it, `emptyLogRecordArguments` takes an argument of the `body` type, so the syntax looks more like `(emptyLogRecordArguments body) {attributes = ..}` which is not as clean but I don't know of a better way to do it.